src/NUnitEngine/nunit.engine.tests/Drivers/NUnit3FrameworkDriverTests.cs

```diff
@@ -149,11 +147,23 @@ public void RunTestsAction_AfterLoad_ReturnsRunnableSuite()
         public void RunTestsAction_WithoutLoad_ThrowsInvalidOperationException()
         {
             var ex = Assert.Catch(() => _driver.Run(new NullListener(), TestFilter.Empty.Text));
-            if (ex is System.Reflection.TargetInvocationException)
+            if (ex is TargetInvocationException)
                 ex = ex.InnerException;
             Assert.That(ex, Is.TypeOf<InvalidOperationException>());
             Assert.That(ex.Message, Is.EqualTo(LOAD_MESSAGE));
         }
+
+        [Test]
+        public void RunTestsAction_WithInvalidFilterElement_ThrowsNUnitEngineException()
+        {
+            _driver.Load(_mockAssemblyPath, _settings);
+
+            var invalidFilter = "<filter><invalidElement>foo</invalidElement></filter>";
+            var ex = Assert.Catch(() => _driver.Run(new NullListener(), invalidFilter));
+            if (ex is TargetInvocationException)
```

Comment:
There aren't any `try catch` blocks between the exception thrown in the Framework and the catching of the TargetInvocationException. [Here](https://github.com/nunit/nunit-console/blob/master/src/NUnitEngine/nunit.engine/Drivers/NUnit3FrameworkDriver.cs#L164-L170) is the relevant `NUnit3FrameworkDriver` code which has no error handling. Is this where you're expecting to see some?

I wonder if we should have something like the following. (Not sure if the null check is required)
```
        private object CreateObject(string typeName, params object[] args)
        {
            try
            {
                return _testDomain.CreateInstanceAndUnwrap(
                    NUNIT_FRAMEWORK, typeName, false, 0,
#if !NET_4_0
                    null, args, null, null, null);
#else
                null, args, null, null );
#endif
            }
            catch (TargetInvocationException ex)
            {
                if (ex.InnerException != null)
                {
                    throw ex.InnerException;
                }
            }
        }
```